dev/devicelab/lib/framework/utils.dart

```diff
@@ -532,10 +544,46 @@ Future<ProcessResult> executeFlutter(String command, {
   List<String> options = const <String>[],
 }) async {
   final List<String> args = _flutterCommandArgs(command, options);
-  return _processManager.run(
+  final ProcessResult processResult = await _processManager.run(
     <String>[path.join(flutterDirectory.path, 'bin', 'flutter'), ...args],
     workingDirectory: cwd,
   );
+
+  if (processResult.exitCode != 0) {
+    await _flutterScreenshot();
+  }
+  return processResult;
+}
+
+Future<void> _flutterScreenshot({ String? workingDirectory }) async {
+  final Directory? dumpDirectory = hostAgent.dumpDirectory;
+  if (dumpDirectory == null) {
+    return;
+  }
+  // On command failure try uploading screenshot of failing command.
```

Comment:
Where does the actual uploading happen? Or do we already upload `hostAgent.dumpDirectory` if it exists?

Reply:
That directory is uploaded as part of the recipe. Anything we copy there is available in the log section of the build.